### PR TITLE
Radio silence and LTE info/metrics

### PR DESCRIPTION
**Pull request by milan-zededa**
Implementation of the Radio-Silence mode and the publishing of cellular info/metrics up to the controller.
This is a follow-up to https://github.com/lf-edge/eve/pull/2314 and https://github.com/lf-edge/eve/pull/2325 (APIs).
For design description see: https://wiki.lfedge.org/display/EVE/EVE+Airplane+Mode
PR consists of two inter-dependent commits - one for the `wwan` service, the other for `zedbox` microservices.
Here are the commit descriptions:

**Refactor wwan service, add radio-silence support**

`wwan` service now works with structured JSON-formatted input/output, which includes config, state data and metrics.
It waits for and reads configuration from `/run/wwan/config.json` and publishes state information
into `status.json`and metrics into `metrics.json` under the same directory.
    
`wwan` service was further improved to:
- allow to reference modem(s) using configurable physical addresses (pci / usb port / interface name)
- support configuration change
- react to configuration change immediately (by watching for `config.json` file changes instead of polling with a long time interval)
- support radio-silence mode (i.e. allow to switch the radio on/off)
- allow to configure IP address for probing
- publish configuration/probing error up to the zedbox
- publish packet stats, modem & SIM card information, operating mode and more
- unify format of published state/metrics data between qmi and mbim protocols
  (i.e. zedbox does not need to know which protocol is being used)
- acknowledge applied configuration with a SHA256 checksum
- prepare support for multiple modems (not tested)

For each network/modem it is still limited to only a single APN.

_Note that at this point the `wwan` service is quite complex, working with structured (JSON formatted)
input/output and it might therefore make sense to re-write it from shell to C or Go for better efficiency.
However, the priority right now is to clearly define API between `wwan` and `zedbox` (config, status, metrics)
and to provide the newly required features like the radio-silence ASAP._

**nim, zedagent: Radio-Silence, wwan status and metrics**

This commit adds Radio-silence feature into zedbox. This covers both cellular modules as well as WiFi adapters.
Radio silence can be temporarily enabled/disabled on demand from a local (profile) server application.
`zedagent` collects and publishes the intended radio state, while `nim` in coordination with the `wwan`
service enables or disables all radio devices and publishes the actual radio state.
The current state of radios is indicated by `ledmanager` and `diag` microservices.
    
Additionally, the behaviour of disabled/unused wireless devices has changed.
Instead of putting such devices under `pciback`, they will remain in `dom0`, but EVE will turn off their radios.
In other words, wireless device without configuration will have RF switched off.
    
Lastly, cellular status and metrics are now collected and published  to the controller
by the `wwan` service -> `nim` -> `zedagent` chain of microservices.

### CODE REVIEW

**Review by rvs — COMMENTED** [review]
Finally got to review this for realz. First of all -- great work @milan-zededa -- really well done!

One question: did we ever update EVE documentation for this? Like LED side and wwan side?

**Comment by milan-zededa** [comment]
> One question: did we ever update EVE documentation for this? Like LED side and wwan side?

Good point, documentation is only on wiki. Will open PR with updated EVE docs.

**Comment by rvs** [comment]
Great @milan-zededa ! Thanks!!!
